nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaLogTailer.java

```diff
@@ -0,0 +1,408 @@
+    }
+
+    protected static String buildSubscribeId(String group, Collection<String> names) {
+        return group + ":" + names.stream().collect(Collectors.joining("|"));
+    }
+
+
+    @Override
+    public LogRecord<M> read(Duration timeout) throws InterruptedException {
+        if (closed) {
+            throw new IllegalStateException("The tailer has been closed.");
+        }
+        if (records.isEmpty()) {
+            int items = poll(timeout);
+            if (isRebalanced) {
+                isRebalanced = false;
+                log.debug("Rebalance happens during poll, raising exception");
+                throw new RebalanceException("Partitions has been rebalanced");
+            }
+            if (items == 0) {
+                if (log.isTraceEnabled()) {
+                    log.trace("No data " + id + " after " + timeout.toMillis() + " ms");
+                }
+                return null;
+            }
+        }
+        ConsumerRecord<String, Bytes> record = records.poll();
+        lastOffsets.put(new TopicPartition(record.topic(), record.partition()), record.offset());
+        M value = messageOf(record.value());
+        LogPartition partition = LogPartition.of(getNameForTopic(record.topic()), record.partition());
+        LogOffset offset = new LogOffsetImpl(partition, record.offset());
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Read from %s/%s, key: %s, value: %s", offset, group, record.key(), value));
+        }
+        return new LogRecord<>(value, offset);
+    }
+
+    protected String getNameForTopic(String topic) {
+        return topic.replaceFirst(prefix, "");
+    }
+
+    @SuppressWarnings("unchecked")
+    protected M messageOf(Bytes value) {
+        ByteArrayInputStream bis = new ByteArrayInputStream(value.get());
+        ObjectInput in = null;
+        try {
+            in = new ObjectInputStream(bis);
+            return (M) in.readObject();
+        } catch (IOException | ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        } finally {
+            try {
+                if (in != null) {
+                    in.close();
+                }
+            } catch (IOException ex) {
+                // ignore close exception
+            }
+        }
+    }
+
+    protected int poll(Duration timeout) throws InterruptedException {
+        records.clear();
+        try {
+            for (ConsumerRecord<String, Bytes> record : consumer.poll(timeout.toMillis())) {
+                if (log.isDebugEnabled() && records.isEmpty()) {
+                    log.debug("Poll first record: " + getNameForTopic(record.topic()) + ":" + record.partition() + ":+" + record.offset());
+                }
+                records.add(record);
+            }
+        } catch (org.apache.kafka.common.errors.InterruptException e) {
+            Thread.currentThread().interrupt();
```

Comment:
`org.apache.kafka.common.errors.InterruptException` constructors already restore the interrupted status with `interrupt()` so no need to redo it.